torch/distributed/fsdp/_optim_utils.py

```diff
@@ -21,14 +23,26 @@
 import torch.distributed.fsdp.fully_sharded_data_parallel as fsdp_file
 import torch.nn as nn
 from torch.distributed._shard.sharded_tensor import ShardedTensor
-from torch.distributed.fsdp._common_utils import _get_param_to_fqns
+from torch.distributed.fsdp._common_utils import (
+    _apply_to_modules,
+    _get_param_to_fqns,
+    _module_handles,
+    clean_tensor_name,
+)
 from torch.distributed.fsdp._fsdp_extensions import _ext_chunk_tensor
 from torch.distributed.fsdp._runtime_utils import _clear_grads_if_needed, _lazy_init
 from torch.distributed.fsdp._shard_utils import _gather_state_dict
 from torch.distributed.fsdp.api import ShardingStrategy
 from torch.distributed.fsdp.flat_param import FlatParameter, FlatParamHandle
 
 
+@dataclass
+class FSDPParamInfo:
+    state: nn.Module
+    flat_param: FlatParameter
+    fqn_indices: Dict[str, int]
```

Comment:
Could we add a comment explaining what these indices mean? (My understanding is that this is a mapping from FQN from `FlatParameter._fqns` to the corresponding index in `FlatParameter._fqns`.)

It also looks like this is not used for this PR (but probably for the next?).

Reply:
Change to `param_indices`.